src/main/java/org/benetech/mathshare/converters/UrlCodeConverterUtils.java

```diff
@@ -0,0 +1,93 @@
+package org.benetech.mathshare.converters;
+
+import java.util.Locale;
+
+public final class UrlCodeConverterUtils {
+
+    private static final int BASE = 32;
+    private static final String MINUS_SIGN = "X";
+    private static final String MIN_LONG_CODE = MINUS_SIGN + "8000000000000";
+
+    public static String toUrlCode(Long number) {
+        validateNumber(number);
+        if (number.equals(Long.MIN_VALUE)) {
```

Comment:
why do we needs this checks, can't we just convert the number to base32, whatever it  is? this seems like too much custom code

Reply:
We can't, because MIN_LONG (-9223372036854775808) can't be simply converted to a positive number (MAX_LONG is 9223372036854775807) and we need it to convert to base32 properly

Reply:
but the base32 output should depend only on the byte array we pass to it. Whether we interpret one of the bits as a sign should be beyond the point.